pizzeria-backend/Controllers/OrderController.cs

```diff
@@ -1,30 +1,134 @@
+        {
+            _orderService = orderService;
+            _hubContext = hubContext;
+        }
 
         [HttpPost()]
-        public IActionResult PostOrder()
+        public async Task<ActionResult> Order([FromBody] OrderDto Order)
+        {
+
+            var ord = await _orderService.MakeOrder(Order);
+
+            if (ord == null)
+            {
+                return BadRequest("One or more of the items are not available or in the menu");
+            }
+
+            string orderString = JsonConvert.SerializeObject(ord, new JsonSerializerSettings
+            {
+                ReferenceLoopHandling = ReferenceLoopHandling.Ignore,
+                Formatting = Formatting.Indented,
+                ContractResolver = new DefaultContractResolver { NamingStrategy = new CamelCaseNamingStrategy() }
+            });
+
+            await _hubContext.Clients.All.SendAsync("NewOrder", orderString);
+            return Ok(ord);
+        }
+
+        [HttpPost("validate")]
+        public async Task<ActionResult> ValidateOrder([FromBody] OrderDto Order)
+        {
+
+            var Ids = await _orderService.ValidateOrder(Order);
+
+            string orderString = JsonConvert.SerializeObject(Ids, new JsonSerializerSettings
+            {
+                ReferenceLoopHandling = ReferenceLoopHandling.Ignore,
+                Formatting = Formatting.Indented,
+                ContractResolver = new DefaultContractResolver { NamingStrategy = new CamelCaseNamingStrategy() }
+            });
+
+            return Ok(new { productIds = Ids.productIds, addOnIds = Ids.addOnIds });
+        }
+
+
+        [HttpGet("{Id}")]
+        public async Task<IActionResult> GetOrder(int Id)
+        {
+            var ord = await _orderService.GetOrder(Id);
+            if (ord == null)
+            {
+                return NotFound("Order not found");
+            }
+
+            return Ok(ord);
+        }
+
+        [HttpDelete("{Id}")]
+        public async Task<IActionResult> DeleteOrder(int Id)
         {
-            return null;
+            var ord = await _orderService.DeleteOrder(Id);
+            if (ord == null)
+            {
+                return NotFound("Order not found");
+            }
+
+            string orderString = JsonConvert.SerializeObject(ord, new JsonSerializerSettings
+            {
+                ReferenceLoopHandling = ReferenceLoopHandling.Ignore,
+                Formatting = Formatting.Indented,
+                ContractResolver = new DefaultContractResolver { NamingStrategy = new CamelCaseNamingStrategy() }
+            });
+
+            await _hubContext.Clients.All.SendAsync("DeleteOrder", orderString);
+            return Ok(orderString);
         }
 
 
-        [HttpGet("{id}")]
-        public IActionResult GetOrder()
+        [HttpGet("change-status/{Id}")]
+        public async Task<IActionResult> ChangeCompletion(int Id)
```

Comment:
by convention the method parameter _id_ should be lowercase

Reply:
Done